### PR TITLE
feat(argo-cd): add support for passing redis - sentinel credentials

### PR DESCRIPTION
**Pull request by shlomitubul**
This PR enables authentication for external Redis with Sentinel by introducing REDIS_SENTINEL_USERNAME and REDIS_SENTINEL_PASSWORD as optional environment variables. These credentials can now be passed via secrets to the controller, repo-server, and server templates, mirroring the approach used for standard external Redis connections.

NOTE:
The actual support for username/password for Sentinel in argo code added in this PR:
https://github.com/argoproj/argo-cd/pull/17168

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by yu-croco** [comment]
> I have updated the chart [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog) with all the changes that come with this pull request according to changelog.

Please handle this.

**Comment by shlomitubul** [comment]
@yu-croco is there anything else that needs to be changed / fixed?

**Comment by pdrastil** [comment]
@shlomitubul PR looks good, we are just waiting on upstream to cherry pick this into release branch

**Comment by corradomatt** [comment]
Any updates on when this might be merged for release?  I'm working on a deployment where I'd like to use Sentinel but it looks like this is a blocker to that.

**Comment by yu-croco** [comment]
Hi @corradomatt .
As discussed https://github.com/argoproj/argo-helm/pull/2492#issuecomment-1973361762 , we are waiting for upstream that https://github.com/argoproj/argo-cd/pull/17168 is released as specific version. 🙋

**Comment by DrFaust92** [comment]
[yu-croco](https://github.com/yu-croco) upstream change was release in 2.11.0, can we push this as well?

**Comment by yu-croco** [comment]
Hi @DrFaust92 , thank you for reminding!
Yes, let's move on 🚀 

Hi @shlomitubul , can you please resume this PR?

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -26,7 +26,4 @@ annotations:
     fingerprint: 2B8F22F57260EFA67BE1C5824B11F800CD9D2252
     url: https://argoproj.github.io/argo-helm/pgp_keys.asc
   artifacthub.io/changes: |
-    - kind: removed
-      description: Remove `server.certificate.secretName`, as the expected secret name is static (argocd-server-tls)
-    - kind: removed
-      description: Remove `applicationSet.certificate.secretName`, as the expected secret name is static (argocd-applicationset-controller-tls)
+      description: Add support for passing redis - sentinel credentials
```
Comment (yu-croco) [inline charts/argo-cd/Chart.yaml:29]: `kind` should be defined. 👀

Comment (shlomitubul) [reply charts/argo-cd/Chart.yaml:30]: Thanks for your review! fixed (:

**Comment by DrFaust92** [comment]
[yu-croco](https://github.com/yu-croco) Hi, can we push this? other things get merged before and we are required playing catch up with chart version.
this is a security feature to be able to use HA redis (sentinel) with authentication.

**Comment by yu-croco** [comment]
Hi @shlomitubul , can you please resolve conflict?

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution. LGTM.

**Review by mbevc1 — APPROVED** [review]

**Comment by DrFaust92** [comment]
thanks [yu-croco](https://github.com/yu-croco)!!!

**Pull request merged**